### PR TITLE
chore: add testing summary to PR template

### PR DESCRIPTION
**Pull request by allenlsy**
Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

Add testing summary section to PR template.

This helps PR authors to inform reviewers how the changes are tested.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [X] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [X] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

No

1. Is this a breaking change?

No

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -35,6 +35,27 @@ Please mark with X for applicable areas.
...
+| Manual Testing              | [ ] |
+| Unit Test                   | [ ] |
+| E2E Test                    | [ ] |
+| Other                       | [ ] |
+
+Any other information about the testing, E.g. screenshots, logs?
+
+
+**Miscellaneous**
+
```
Comment (shashankram) [inline .github/pull_request_template.md:58]:
```suggestion
```

I think `Testing done:` should suffice.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -35,9 +35,20 @@ Please mark with X for applicable areas.
...
 Please answer the following questions with yes/no.
 
 1. Does this change contain code from or inspired by another project?
     -   Did you notify the maintainers and provide attribution?
 
-1. Is this a breaking change?
+2. Is this a breaking change?
```
Comment (shashankram) [inline .github/pull_request_template.md:54]: Not needed. Markdown will correctly render this as 2.

File: .github/pull_request_template.md (hunk 1 above)
Comment (allenlsy) [reply .github/pull_request_template.md]: Updated.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -35,9 +35,20 @@ Please mark with X for applicable areas.
...
+**Testing done**:
+
+<!--
+
+Please describe how are the changes tested? You can add supporting information, E.g. screenshots, logs etc.
+
+-->
+
+
+**Miscellaneous**
```
Comment (shashankram) [inline .github/pull_request_template.md:47]: Not sure if we need this really. Let's keep it simple and not have it unless there is a need. In general Miscellaneous comments can be regular GH comments.

File: .github/pull_request_template.md (hunk 2 above)
Comment (allenlsy) [reply .github/pull_request_template.md:54]: Didn't mean to change it. It's updated by my editor. Will revert

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -35,9 +35,19 @@ Please mark with X for applicable areas.
...
+**Testing done**:
+
+<!--
+
+Please describe how are the changes tested? You can add supporting information, E.g. screenshots, logs etc.
+
+-->
+
+
+
```
Comment (shashankram) [inline .github/pull_request_template.md:47]:
```suggestion
**Testing done**:
<!--

Please describe how are the changes tested? You can add supporting information, E.g. screenshots, logs etc.

-->


```

**Review by shashankram — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**